File: backend/src/main/java/be/kauffman/kfm/modules/user/entity/payload/UserCreatePayload.java
package be.kauffman.kfm.modules.user.entity.payload;

import be.kauffman.kfm.modules.site.entity.dto.Site;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserCreatePayload {
    String firstname;
    String lastname;
    String gender;
    String avatar;
    Date dob;
    String email;
    String password;
    String telpro;
    String telperso;
    String nationality;
    String numirn;
    String driver_license;
    Date created_on;
    String pob;
    String active;
    Site site;
}

File: frontend/src/app/module/user/model/payload/user-create.payload.ts
import {PayloadInterface} from '@shared/model';

export interface UserCreatePayload extends PayloadInterface {
  firstname: string;
  lastname: string;
  gender: string;
  avatar: string;
  dob: Date;
  email: string;
  password: string;
  telpro: string;
  telperso: string;
  nationality: string;
  numirn: string;
  driver_license: string;
  created_on: Date;
  pob: string;
  active: string;
  site: any;
  status: any;
  //grade: any;
  //address: any;
}
